src/machine.rs

```diff
@@ -561,7 +571,26 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'mir, 'tcx> {
         let call_id = stacked_borrows.map_or(NonZeroU64::new(1).unwrap(), |stacked_borrows| {
             stacked_borrows.borrow_mut().new_call()
         });
-        let extra = FrameData { call_id, catch_unwind: None };
+        let timing = if let Some(profiler) = ecx.machine.profiler.as_ref() {
+            let fn_name = frame.instance.to_string();
+            let entry = ecx.machine.string_cache.entry(fn_name.clone());
+            let name = match entry {
+                Entry::Occupied(e) => *e.get(),
+                Entry::Vacant(e) => {
+                    *e.insert(profiler.alloc_string(&*fn_name))
+                }
+            };
+
+            Some(profiler.start_recording_interval_event_detached(
```

Comment:
Below you are doing the `finish_recording_interval_event` *after* `ecx.handle_stack_pop`... so for consistency, would it make sense to do the "start" *before* anything else in this function?